pilot/pkg/security/authz/model/generator_test.go

```diff
@@ -542,7 +542,7 @@ func TestServiceAccount(t *testing.T) {
 	}
 	for _, tt := range cases {
 		t.Run(tt.Name, func(t *testing.T) {
-			r, _ := serviceAccountRegex(input)
+			r := serviceAccountRegex("", input)
```

Comment:
Do we want to add a test case for when the input is of the form "my-sa" and we provide a default namespace?

Reply:
Add a new test, not a test case -- the test here is a bit abnormal, its calling serviceAccountRegex once with a static input to assert the regex meets what we expect. Ill add a new one with a more typical test setup of calling serviceAccountRegex with various inputs now that the function is not trivial